File: skyi-monitor/microservices/alert-service/src/main/java/com/skyi/alert/model/Alert.java
package com.skyi.alert.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;

/**
 * 告警实体类
 */
@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "t_alert")
public class Alert {

    /**
     * 告警ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 告警UUID
     */
    @Column(length = 64, unique = true, nullable = false)
    private String alertUuid;

    /**
     * 告警名称
     */
    @Column(length = 200, nullable = false)
    private String name;

    /**
     * 告警内容
     */
    @Column(length = 1000)
    private String message;

    /**
     * 告警级别：CRITICAL（严重）、WARNING（警告）、INFO（提示）
     */
    @Column(length = 20, nullable = false)
    @Enumerated(EnumType.STRING)
    private AlertSeverity severity;

    /**
     * 告警状态：ACTIVE（未处理）、ACKNOWLEDGED（已确认）、RESOLVED（已解决）、CLOSED（已关闭）
     */
    @Column(length = 20, nullable = false)
    @Enumerated(EnumType.STRING)
    private AlertStatus status;

    /**
     * 告警类型：THRESHOLD（阈值）、TREND（趋势）、PATTERN（模式）
     */
    @Column(length = 20, nullable = false)
    @Enumerated(EnumType.STRING)
    private AlertType type;

    /**
     * 资产ID
     */
    @Column(nullable = false)
    private Long assetId;

    /**
     * 资产名称
     */
    @Column(length = 200)
    private String assetName;

    /**
     * 资产类型
     */
    @Column(length = 100)
    private String assetType;

    /**
     * 指标名称
     */
    @Column(length = 100)
    private String metricName;

    /**
     * 指标值
     */
    private Double metricValue;

    /**
     * 阈值
     */
    private Double threshold;

    /**
     * 告警开始时间
     */
    @Column(nullable = false)
    private LocalDateTime startTime;

    /**
     * 告警结束时间
     */
    private LocalDateTime endTime;

    /**
     * 告警确认时间
     */
    private LocalDateTime acknowledgedTime;

    /**
     * 告警确认人
     */
    @Column(length = 100)
    private String acknowledgedBy;

    /**
     * 告警解决时间
     */
    private LocalDateTime resolvedTime;

    /**
     * 告警解决人
     */
    @Column(length = 100)
    private String resolvedBy;

    /**
     * 告警处理备注
     */
    @Column(length = 1000)
    private String resolveComment;

    /**
     * 是否已通知
     */
    private Boolean notified;

    /**
     * 上次通知时间
     */
    private LocalDateTime lastNotifiedTime;

    /**
     * 通知次数
     */
    private Integer notificationCount;

    /**
     * 告警标签，JSON格式
     */
    @Column(columnDefinition = "TEXT")
    private String tags;

    /**
     * 创建时间
     */
    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @UpdateTimestamp
    private LocalDateTime updateTime;
} 

File: skyi-monitor/frontend/src/types/alert.ts
/**
 * 告警通知接口
 */
export interface AlertNotification {
  id?: number;
  alertId: number;
  alertUuid: string;
  channel: NotificationChannel;
  recipient?: string;
  content?: string;
  status: NotificationStatus;
  failReason?: string;
  retryCount?: number;
  sentTime?: string;
  createTime?: string;
  updateTime?: string;
}

/**
 * 通知渠道枚举
 */
export enum NotificationChannel {
  EMAIL = 'EMAIL',
  SMS = 'SMS',
  WEBHOOK = 'WEBHOOK',
  DINGTALK = 'DINGTALK',
  WEIXIN = 'WEIXIN'
}

/**
 * 通知状态枚举
 */
export enum NotificationStatus {
  PENDING = 'PENDING',
  SUCCESS = 'SUCCESS',
  FAILED = 'FAILED'
}

/**
 * 告警级别枚举
 */
export enum AlertSeverity {
  CRITICAL = 'CRITICAL',
  MAJOR = 'MAJOR',
  MINOR = 'MINOR',
  WARNING = 'WARNING',
  INFO = 'INFO'
}

/**
 * 告警状态枚举
 */
export enum AlertStatus {
  ACTIVE = 'ACTIVE',
  ACKNOWLEDGED = 'ACKNOWLEDGED',
  RESOLVED = 'RESOLVED',
  EXPIRED = 'EXPIRED'
}

/**
 * 告警类型枚举
 */
export enum AlertType {
  THRESHOLD = 'THRESHOLD',
  TREND = 'TREND',
  PATTERN = 'PATTERN'
}

/**
 * 告警接口
 */
export interface Alert {
  id?: number;
  uuid: string;
  name: string;
  message: string;
  severity: AlertSeverity;
  status: AlertStatus;
  type: AlertType;
  assetId: number;
  assetName: string;
  assetType: string;
  metricName: string;
  metricValue?: number;
  threshold?: number;
  notified: boolean;
  acknowledgedBy?: string;
  acknowledgedTime?: string;
  resolvedBy?: string;
  resolvedTime?: string;
  resolveComment?: string;
  createTime?: string;
  updateTime?: string;
}

/**
 * 告警查询参数接口
 */
export interface AlertQueryParams {
  status?: string;
  severity?: string;
  assetId?: number;
  assetType?: string;
  startTime?: string;
  endTime?: string;
  page: number;
  size: number;
}

/**
 * 分页结果接口
 */
export interface PaginationResult<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  empty: boolean;
}

/**
 * API响应接口
 */
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
} 